homeworks/vulmanjanos/9_fastapi/models.py

```diff
@@ -0,0 +1,29 @@
+from sqlalchemy import Column, String, Integer, Float
+from pydantic import BaseModel
+from typing import Optional  # Importáljuk az Optional típust
+from ws_database import Base
+
+# Adatbázis modell
+class Item(Base):
+    __tablename__ = "items"
+    id = Column(String, primary_key=True, index=True)  # Primary key beállítása
```

Comment:
A post request-ed HTTP 500-as internal server error-t fog dobni, mert nem hozod létre automatikusan az id-t (ezt ugye nem a felhasználó adja meg).

próbáld így:

```
import uuid
from sqlalchemy.dialects.postgresql import UUID
# Adatbázis modell
class Item(Base):
    __tablename__ = "items"
    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4())) # Primary key beállítása
    item_name = Column(String, index=True)
    quantity = Column(Integer)
    price = Column(Float)
    category = Column(String, index=True, nullable=True)  # Opcionális mező
```

Reply:
ez még hiányzik, hogy adj egy default értéket az id-nak minden db-be írásnál.